### PR TITLE
Add test cases for "UnicodeToLatex" and "LatexToUnicode"

### PR DESCRIPTION
**Pull request by AbdAlRahmanGad**
I added the test case mentioned in [#5547](https://github.com/JabRef/jabref/issues/5547#issuecomment-554067764) and other test cases to ensure reliability. 

The problem with the test case was that ```ı̄``` is not one character it's a combination of ```ı  +   ̄``` unlike ```ā```. What was happennig is that ```ı``` was being converted to ```\i``` and then ```  ̄ ``` was being converted to ```{\={}}``` so the result would be ```{\i{\={}}}```.

I made it that we deal with such characters that cause conflict after combining accents. They are stored in a new variable called ```UNICODE_LATEX_CONVERSION_MAP_AFTER_COMBINING_ACCENTS``` I think this will be the base to deal with such cases.

I also added some of the underdot characters as the previous implementation didn't handle them.

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by calixtus** [comment]
Maybe this could help: https://www.unicode.org/faq/char_combmark.html

**Comment by koppor** [comment]
I think, all the unicode should be normalized before conversion. Use the formatter introduced at https://github.com/JabRef/jabref/pull/11056.

File: src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java [hunk 1]
```diff
@@ -20,6 +20,12 @@ void formatUmlaut() {
         assertEquals("Ä", formatter.format("{\\\"{A}}"));
     }
 
+    @Test
+    void smallIwithoutDot() {
+        assertEquals("ı", formatter.format("\\i"));
+        assertEquals("ı", formatter.format("{\\i}"));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java:26]: Make a `@ParamterizedTest`. I think `@CsvSource` will work well.

**Comment by koppor** [comment]
> I made it that we deal with such characters that cause conflict after combining accents.

My comment above in other words: We need to rely on the normal form NFC. Base our internal maps on that. And do not introduce some other maps.

File: src/main/java/org/jabref/logic/formatter/bibtexfields/UnicodeToLatexFormatter.java [hunk 2]
```diff
@@ -23,6 +23,9 @@ public String format(String text) {
             return result;
         }
 
+        NormalizeUnicodeFormatter normalizer = new NormalizeUnicodeFormatter();
```
Comment (koppor) [inline src/main/java/org/jabref/logic/formatter/bibtexfields/UnicodeToLatexFormatter.java:26]: Make this a class constant: `public static final ...`.

Open the context menu, select "Refactor..." then "Introduce Field..."

image

Choose "field declaration" for initialization:

image

You will then have to make the variable name upper case. You can use <kbd>Alt</kbd>+<kbd>Enter</kbd> to get IntelliJ support:

image

File: src/main/java/org/jabref/logic/util/strings/HTMLUnicodeConversionMaps.java [hunk 3]
```diff
@@ -16,7 +16,6 @@ public class HTMLUnicodeConversionMaps {
     public static final Map<String, String> UNICODE_LATEX_CONVERSION_MAP = new HashMap<>();
     public static final Map<String, String> LATEX_HTML_CONVERSION_MAP = new HashMap<>();
     public static final Map<String, String> LATEX_UNICODE_CONVERSION_MAP = new HashMap<>();
-
```
Comment (koppor) [inline src/main/java/org/jabref/logic/util/strings/HTMLUnicodeConversionMaps.java:19]: Please keep the empty line as separator

File: src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java [hunk 4]
```diff
@@ -20,6 +22,16 @@ void formatUmlaut() {
         assertEquals("Ä", formatter.format("{\\\"{A}}"));
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "ı, \\i",
+            "ı, {\\i}"
+    })
+
```
Comment (koppor) [inline src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java:30]: Remove the empty line

File: src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java [hunk 5]
```diff
@@ -163,6 +175,11 @@ void conversionOfUnderscoreWithBraces() {
         assertEquals("Lorem ipsum_(lorem ipsum)", formatter.format("Lorem ipsum_{lorem ipsum}"));
     }
 
+    @Test
+    void twoDifferentMacrons() {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java:179]: Please add the a link to the issue. Just as JavaDoc with `<a href="...">...`

**Review by koppor — CHANGES_REQUESTED** [review]
Thank you for the quick action taken. Just minor comments.

---

Future work: The part at " // Combining accents" will never be touched? One should check with "code coverage"

**Review by koppor — APPROVED** [review]

**Pull request merged**